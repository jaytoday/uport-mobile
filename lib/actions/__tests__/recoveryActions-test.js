// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  showRecoverySeed,
  hideRecoverySeed,
  saveRecoverySeed,
  selectRecoveryWord,
  startSeedRecovery
 } from 'uPortMobile/lib/actions/recoveryActions'

it('creates a SHOW_RECOVERY_SEED action', () => {
  expect(showRecoverySeed()).toMatchSnapshot()
})

it('creates a SAVE_RECOVERY_SEED action', () => {
  expect(saveRecoverySeed('secret seed')).toMatchSnapshot()
})

it('creates a HIDE_RECOVERY_SEED action', () => {
  expect(hideRecoverySeed()).toMatchSnapshot()
})

it('creates a SELECT_RECOVERY_SEED_WORD action', () => {
  expect(selectRecoveryWord(8)).toMatchSnapshot()
})

it('creates a START_SEED_RECOVERY action', () => {
  expect(startSeedRecovery('some secret phase', 123)).toMatchSnapshot()
})
