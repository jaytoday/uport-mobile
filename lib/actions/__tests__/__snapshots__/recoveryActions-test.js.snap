// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a HIDE_RECOVERY_SEED action 1`] = `
Object {
  "type": "HIDE_RECOVERY_SEED",
}
`;

exports[`creates a SAVE_RECOVERY_SEED action 1`] = `
Object {
  "seed": "secret seed",
  "type": "SAVE_RECOVERY_SEED",
}
`;

exports[`creates a SELECT_RECOVERY_SEED_WORD action 1`] = `
Object {
  "index": 8,
  "type": "SELECT_RECOVERY_SEED_WORD",
}
`;

exports[`creates a SHOW_RECOVERY_SEED action 1`] = `
Object {
  "type": "SHOW_RECOVERY_SEED",
}
`;

exports[`creates a START_SEED_RECOVERY action 1`] = `
Object {
  "componentId": 123,
  "phrase": "some secret phase",
  "type": "START_SEED_RECOVERY",
}
`;
